[sparsity][refactor] Restructure the tests folders

z-a-f opened the pull request:
Summary: There will be more sparse tests coming. This PR creates a separate folder for the sparse tests

Test Plan: `python test/test_ao.py`

Differential Revision: D29139265

raghuramank100 commented on test/test_ao.py:
```diff
@@ -0,0 +1,12 @@
+# -*- coding: utf-8 -*-
+
+from torch.testing._internal.common_utils import run_tests
```
Should we call this file test_ao_sparsity.py so that we eventually maintain separate top level files for quantization and sparsity?

z-a-f replied:
Sure, that's not a bad plan

Commit:
[sparsity][refactor] Restructure the tests folders (#60032)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/60032

There will be more sparse tests coming. This PR creates a separate folder for the sparse tests

Test Plan: `python test/test_ao.py`

Reviewed By: raghuramank100

Differential Revision: D29139265

fbshipit-source-id: 64dd0e8b37a9684bb79a43a0dfd9ec7970631afc
